store/tikv/2pc.go

```diff
@@ -924,7 +959,8 @@ func (c *twoPhaseCommitter) cleanup(ctx context.Context) {
 func (c *twoPhaseCommitter) execute(ctx context.Context) (err error) {
 	var binlogSkipped bool
 	defer func() {
-		if !c.isAsyncCommit() {
+		isOnePC := c.isOnePC()
+		if !c.isAsyncCommit() && !isOnePC {
```

Comment:
How about 
```
if c.isAsyncCommit() {
} else if isOnePC {
} else some other branches {
} else {
}
```
Thus we may not miss some branches.
Or should we do some refactor on this `execute` function.